### PR TITLE
test: run `test_flush_reopen_region` and `test_region_replay` with `KafkaLogStore`

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Run `test_flush_reopen_region` and `test_region_replay` with `KafkaLogStore`

Prelude of #4075

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/mito2/Cargo.toml [hunk 1]
```diff
@@ -54,6 +55,9 @@ prost.workspace = true
 puffin.workspace = true
 rand.workspace = true
 regex = "1.5"
+rskafka = { workspace = true, optional = true }
+rstest = { version = "0.20", optional = true }
+rstest_reuse = { version = "0.7", optional = true }
```
Comment (evenyag) [inline src/mito2/Cargo.toml:60]: Could we add them to workspace dependencies?

File: src/mito2/src/engine/basic_test.rs [hunk 2]
```diff
@@ -83,14 +88,26 @@ async fn test_write_to_region() {
...
     let region_id = RegionId::new(1, 1);
-    let request = CreateRequestBuilder::new().build();
+    let mut request = CreateRequestBuilder::new().build();
+    let topic_and_options = prepare_test_for_kafka_log_store(&factory).await;
+    if let Some((_, wal_options)) = &topic_and_options {
+        request.options.insert(
+            WAL_OPTIONS_KEY.to_string(),
+            serde_json::to_string(&wal_options).unwrap(),
+        );
+    };
```
Comment (evenyag) [inline src/mito2/src/engine/basic_test.rs:109]: Could we do it in the CreateRequestBuilder by setting the factory to the builder?

Comment (WenyXu) [reply src/mito2/src/engine/basic_test.rs:109]: Good point

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**